lib/rspec/matchers/matcher_delegator.rb

```diff
@@ -1,8 +1,34 @@
 module RSpec
   module Matchers
+    # Provides a base class with as little methods as possible, so that
+    # most methods can be delegated via `method_missing`.
+    #
+    # On Ruby 2.0+ BasicObject could be used for this purpose, but it
+    # introduce some extra complexity with constant resolution, so the
+    # BlankSlate pattern was prefered.
+    # @private
+    class BaseDelegator
+      kept_methods = [
+        # Methods that raise warnings if removed.
+        :__id__, :__send__, :object_id,
+
+        # Methods that are explicitly undefined in some subclasses.
+        :==, :===,
```

Comment:
So I had to keep these two only to not break:

```ruby
    # Decorator used for matchers that have special implementations of
    # operators like `==` and `===`.
    # @private
    class AliasedMatcherWithOperatorSupport < AliasedMatcher
      # We undef these so that they get delegated via `method_missing`.
      undef ==
      undef ===
    end
```

I think we could remove them from `BaseDelegator` and just make `AliasedMatcherWithOperatorSupport` and empty class.

Reply:
`===` is the base method matchers use to match on, so if they are explicitly undefined its likely to fall back to delegated ones

Reply:
What I meant by this comment, is that I left them in the BlankSlate to minimize the diff.

But IMO they should be removed from the BlankSlate and `AliasedMatcherWithOperatorSupport` should stop undefining them.

Reply:
https://github.com/rspec/rspec-expectations/commit/0b13e898d878d3dfde622f63956fb32c77469c0c